lego/apps/surveys/views.py

```diff
@@ -68,3 +74,45 @@ def create(self, request, *args, **kwargs):
         return Response(
             SubmissionReadSerializer(serializer.instance).data, status=status.HTTP_201_CREATED
         )
+
+    def validate_answer(self, request, **kwargs):
+        submission = Submission.objects.get(pk=kwargs['pk'])
+        answer_pk = request.query_params.get('answer')
+        if answer_pk is None:
+            raise exceptions.NotAcceptable('No answer specified')
+        try:
+            answer = submission.answers.get(pk=answer_pk)
+        except Answer.DoesNotExist:
+            raise exceptions.NotFound('Answer not found')
+        if answer.question.question_type != TEXT_FIELD:
+            raise exceptions.NotAcceptable('Only text answers can be hidden')
+        return submission, answer
+
+    @detail_route(methods=['GET'])
+    def hide(self, request, **kwargs):
+        submission, answer = self.validate_answer(request, **kwargs)
+        answer.hide()
+        return Response(
+            data=SubmissionReadSerializer(submission).data, status=status.HTTP_202_ACCEPTED
+        )
+
+    @detail_route(methods=['GET'])
+    def show(self, request, **kwargs):
+        submission, answer = self.validate_answer(request, **kwargs)
+        answer.show()
+        return Response(
+            data=SubmissionReadSerializer(submission).data, status=status.HTTP_202_ACCEPTED
+        )
+
+
+class SurveyTokenViewset(viewsets.GenericViewSet):
```

Comment:
Since `token` in Survey is `unique=True`, do you need a `viewset` then? There should be no need to supply the pk of the survey. The only thing used is `GET` on a single endpoint (more or less), so a simple [view](http://www.django-rest-framework.org/api-guide/views/) should be more then enough, or?